src/modules/previewpane/powerpreview/powerpreview.cpp

```diff
@@ -218,56 +230,59 @@ void PowerPreviewModule::apply_settings(const PowerToysSettings::PowerToyValues&
 {
     bool notifyShell = false;
 
-    for (auto& fileExplorerModule : m_fileExplorerModules)
+    if (m_enabled)
```

Comment:
@htcfreek
For point 5, I added a check that should prevent any preview handlers from running if the module is not enabled.

Reply:
Sure. But the policy state won't be written into the settings file. You have to explicit query the policy values and evaluate them.

Reply:
Because you shouldn't be able to bypass a force-enabled add-on by turning off all extensions?

Reply:
But what about force disabled?

Reply:
What do you think of `powerpreview.cpp` now?

Reply:
yes, it does look good to me now.